tests/test_enhanced_sampling.py

```diff
@@ -91,4 +103,207 @@ def get_torsion(x_t):
+    # plt.show()
+
+    assert np.abs(np.average(vacuum_torsions, weights=vacuum_weights)) < 0.1
+
+    num_batches = 10000
+
+    frozen_masses = np.copy(masses)
+    frozen_masses[-num_ligand_atoms:] = np.inf
+
+    all_torsions = []
+
+    # test with both frozen masses and free masses
+    for test_masses in [frozen_masses, masses]:
+
+        sample_generator = enhanced.generate_solvent_phase_samples(
+            ubps, params, test_masses, coords, box, temperature, num_batches=num_batches
+        )
+
+        next_x = None
+
+        K = 200
+
+        mover = AlignedMover(K, vacuum_samples, vacuum_weights, log_prob_fn)
+
+        enhanced_torsions = []
+        for iteration in range(num_batches):
+
+            x_solvent, v_solvent, box_solvent = sample_generator.send(next_x)
+            solvent_torsion = get_torsion(x_solvent[-num_ligand_atoms:])
+            enhanced_torsions.append(solvent_torsion)
+
+            next_x = mover.move(CoordsVelBox(x_solvent, v_solvent, box_solvent))
+
+            print(
+                f"frame {iteration} accepted {mover.n_accepted} proposed {mover.n_proposed} solvent_torsion {solvent_torsion}"
+            )
+
+        all_torsions.append(enhanced_torsions)
+
+    all_torsions = np.asarray(all_torsions)
+
+    # (ytz): useful for visualization, so please leave this comment here.
+    from matplotlib import pyplot as plt
+
+    # plt.hist(vanilla_solvent_torsions, bins=np.linspace(-np.pi, np.pi, 100), density=True, alpha=0.5)
+
+    enhanced_torsions_lhs, _ = np.histogram(enhanced_torsions, bins=50, range=(-np.pi, 0), density=True)
+    enhanced_torsions_rhs, _ = np.histogram(enhanced_torsions, bins=50, range=(0, np.pi), density=True)
+
+    # check for symmetry about theta=0
+    assert np.mean((enhanced_torsions_lhs - enhanced_torsions_rhs[::-1]) ** 2) < 5e-2
+
+    vanilla_torsions = []
+    # generate reference samples:
+    for x_solvent, box_solvent in enhanced.generate_solvent_phase_samples(
+        ubps, params, masses, coords, box, temperature, num_batches=num_batches
+    ):
+        solvent_torsion = get_torsion(x_solvent[-num_ligand_atoms:])
+        vanilla_torsions.append(solvent_torsion)
+
+    vanilla_torsions = np.asarray(vanilla_torsions)
+
+    plt.hist(all_torsions[0], bins=np.linspace(-np.pi, np.pi, 100), density=True, alpha=0.5, label="enhanced_frozen")
+    plt.hist(all_torsions[1], bins=np.linspace(-np.pi, np.pi, 100), density=True, alpha=0.5, label="enhanced_free")
+    plt.hist(vanilla_torsions, bins=np.linspace(-np.pi, np.pi, 100), density=True, alpha=0.25, label="vanilla")
+    plt.legend()
+    plt.xlabel("torsion angle")
+    plt.ylabel("density")
+    plt.show()
+
+    vanilla_samples_lhs, _ = np.histogram(vanilla_torsions, bins=50, range=(-np.pi, 0), density=True)
+
+    # check for consistency with vanilla samples
+    assert np.mean((enhanced_torsions_lhs - vanilla_samples_lhs) ** 2) < 5e-2
+
+
+if __name__ == "__main__":
+    test_condensed_phase_mtm()
```

Comment:
Is this to imply this test shouldn't be part of the regular test suite?

Reply:
I will remove this in a final pass!